File: src/test/java/no/fintlabs/repository/utils/performance/Timer.java
package no.fintlabs.repository.utils.performance;

import lombok.extern.slf4j.Slf4j;

import java.time.Duration;

@Slf4j
public class Timer {

    private final Long startTime;

    private Timer(Long startTime) {
        this.startTime = startTime;
    }

    public static Timer start() {
        return new Timer(System.nanoTime());
    }

    public Duration getElapsedTime() {
        return Duration.ofNanos(System.nanoTime() - startTime);
    }

}

File: src/test/kotlin/no/novari/flyt/history/repository/utils/performance/Timer.kt
package no.novari.flyt.history.repository.utils.performance

import java.time.Duration

class Timer private constructor(
    private val startTime: Long,
) {
    val elapsedTime: Duration
        get() = Duration.ofNanos(System.nanoTime() - startTime)

    companion object {
        @JvmStatic
        fun start() = Timer(System.nanoTime())
    }
}
